src/aten/sycl/UpSampleBicubic2dKernels.cpp

```diff
@@ -0,0 +1,184 @@
+#include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
+#include <ATen/Dispatch.h>
+#include <ATen/TensorUtils.h>
+#include <comm/SYCLContext.h>
+#include "UpSample.h"
+
+namespace at::native::xpu {
+
+template <typename scalar_t, typename accscalar_t>
+struct UpsampleBicubic2dOutFrameKernelFunctor {
+  void operator()(sycl::nd_item<1> item) const {
+    auto idata = in_data_;
+    auto odata = out_data_;
+    int global_id = item.get_global_linear_id();
+    const int nbatch = idata.size(0);
+    const int channels = idata.size(1);
+    const int input_height = idata.size(2);
+    const int input_width = idata.size(3);
+    const int output_height = odata.size(2);
+    const int output_width = odata.size(3);
+    if (global_id < output_height * output_width) {
+      const int output_x = global_id % output_width;
+      const int output_y = global_id / output_width;
+      if (input_height == output_height && input_width == output_width) {
+        for (int n = 0; n < nbatch; n++) {
+          for (int c = 0; c < channels; c++) {
+            auto val = idata[n][c][output_y][output_x];
+            odata[n][c][output_y][output_x] = val;
+          }
+        }
+        return;
+      }
+
+      // Interpolation kernel
+      accscalar_t real_x = area_pixel_compute_source_index(
+          width_scale_, output_x, align_corners_, /*cubic=*/true);
+      // TODO: floor should support at dispatch to half type
```

Comment:
Can you evaluate the issue?

Reply:
> Can you evaluate the issue?

already used c++ standard floorf(), follow cuda, forget to remove this comment